src/components/NewMovie/NewMovie.js

```diff
@@ -1,13 +1,90 @@
 import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import { validateForm } from './validateForm';
+import { Input } from './Input';
+
+const movies = [
+  'title',
+  'description',
+  'imgUrl',
+  'imdbUrl',
+  'imdbId',
+];
+
+const initialValues = movies.reduce((acc, name) => {
+  return {
+    ...acc,
+    [name]: '',
+  };
+}, {});
+
+const initialErrors = movies.reduce((acc, name) => {
+  return {
+    ...acc,
+    [name]: null,
+  };
+}, {});
 
 export class NewMovie extends Component {
-  state = {};
+  state = {
+    values: initialValues,
+    errors: initialErrors,
+  };
+
+  handleChange = ({ target }) => {
+    this.setState(prevState => ({
+      values: {
+        ...prevState.values,
+        [target.name]: target.value,
+      },
+    }));
+  };
+
+  handleSubmit = (event) => {
+    event.preventDefault();
+    const { onSubmit } = this.props;
+
+    this.setState((prevState) => {
+      const {
+        errors,
+        hasErrors,
+      } = validateForm(prevState.values);
+
+      if (hasErrors) {
+        return {
+          errors,
+        };
+      }
+
+      onSubmit(prevState.values);
+
+      return {
+        errors,
+        values: initialValues,
+      };
+    });
+  };
 
   render() {
+    const { values, errors } = this.state;
+
     return (
-      <form>
-        Put the form here
+      <form className="ui form" onSubmit={this.handleSubmit}>
+        {Object.entries(values).map(([movie, value]) => (
+          <Input
+            key={movie}
+            movie={movie}
+            values={value}
```

Comment:
Тут має бути в однині
```suggestion
            value={value}
```